File: Compliance Quality Assurance and Tests/MottuFlow/src/main/java/com/sprint/MottuFlow/domain/arucotag/ArucoTag.java
package com.sprint.MottuFlow.domain.arucotag;

import com.sprint.MottuFlow.domain.moto.Moto;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "aruco_tag")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ArucoTag {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idTag;

    @Column(nullable = false, length = 50)
    private String codigo;

    @Column(nullable = false, length = 20)
    private String status;

    @ManyToOne
    @JoinColumn(name = "id_moto", nullable = false)
    private Moto moto;

	public Long getIdTag() {
		return idTag;
	}

	public void setIdTag(Long idTag) {
		this.idTag = idTag;
	}

	public String getCodigo() {
		return codigo;
	}

	public void setCodigo(String codigo) {
		this.codigo = codigo;
	}

	public Moto getMoto() {
		return moto;
	}

	public void setMoto(Moto moto) {
		this.moto = moto;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}
}

File: Advanced Business Development With dotNET/Models/ArucoTag.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MottuFlowApi.Models
{
    [Table("aruco_tag")]
    public class ArucoTag
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id_tag")]
        public int IdTag { get; set; }

        [Required]
        [MaxLength(50)]
        [Column("codigo")]
        public string Codigo { get; set; } = string.Empty;

        [Required]
        [MaxLength(20)]
        [Column("status")]
        public string Status { get; set; } = string.Empty;

        [Required]
        [Column("id_moto")]
        public int IdMoto { get; set; }

        // Navigation
        public Moto? Moto { get; set; }
    }
}
